lalrpop/Cargo.toml

```diff
@@ -15,23 +15,64 @@ exclude = ["build.rs"]
 [lib]
 doctest = false
 
-[dependencies]
-ascii-canvas = "2.0"
-atty = "0.2"
-bit-set = "0.5"
-diff = "0.1.12"
-docopt = "1"
-ena = "0.14"
-itertools = "0.9"
-regex = "1"
-regex-syntax = "0.6"
-petgraph = "0.5"
-serde = "1.0"
-serde_derive = "1.0"
-string_cache = "0.8"
-term = "0.5"
-unicode-xid = "0.2"
-sha2 = "0.8"
+[dependencies.ascii-canvas]
+version = "2.0"
+default_features = false
+
+[dependencies.atty]
+version = "0.2"
+default_features = false
+
+[dependencies.bit-set]
+version = "0.5"
+default_features = false
+
+[dependencies.diff]
+version = "0.1.12"
+default_features = false
+
+[dependencies.pico-args]
```

Comment:
I think pico-args could be an optional dependency as it isn't needed when using lalrpop as a library.

Reply:
Had to move the CLI binary into a separate workspace, not sure how that affects downstream users. I suggest we publish it separately for those that want to use it, which I guess is the minority as most people probably only care about the library part. Even though now with pico-args the compile time impact is minimal.

Reply:
You should not need to do that. Make the dependency optional, use `required-features` for the bin target and then add `pico-args` to the default feature (to reduce breakage).

Reply:
Why would you need to move the binary out? Why not just make the dependency optional and make it a `required-features` for that binary?

Reply:
feature + optional, tried that, didn't work. It requires more, some way of cfg-ing the dependency out to only show up for binary builds. And I couldn't find a way to do so. Please feel free to push to this PR yourself if you have further suggestion how this specific issue should be resolved, as I'm out of my depth.

Reply:
Yes, there is unfortunately no way to force the feature enabled on binary build but by using `required-features` anyone doing `cargo install` will get a helpful error pointing out the missing feature. Alternately `pico-args` could bemade default and a `default-features = false` will disable it easily for lib builds.

I think either of those solutions is good enough, and easier than using  separate crate.

https://doc.rust-lang.org/cargo/reference/cargo-targets.html#binaries